src/models/mod.rs

```diff
@@ -36,19 +39,40 @@ const MIGRATIONS: EmbeddedMigrations = embed_migrations!("migrations");
 
 pub type DBConnection = PooledConnection<ConnectionManager<PgConnection>>;
 
+pub enum DBConnectionType<'a> {
+  #[allow(dead_code)]
+  Test,
+  Normal {
+    channel_name: &'a str,
+  },
+}
+
 pub struct DBPool {
   inner_pool: Pool<ConnectionManager<PgConnection>>,
 }
 
+fn get_channel_db_url<'a>(conn_type: &DBConnectionType<'a>) -> String {
```

Comment:
I think lifetime elision might be smart enough to figure out the annotations here.